File: matlab/ch02/ex234.m
% cp1dintp.m: 1-D interpolation for Cp data 
T  =  373:100:873; Cp  =  [29.189 29.291 29.462 29.678 29.971 30.269]; % data 
lCp  =  interp1(T, Cp, 580, 'linear'); pCp  =  interp1(T, Cp, 580, 'pchip'); 
sCp  =  interp1(T, Cp, 580, 'spline'); nCp  =  interp1(T, Cp, 580, 'nearest'); 
fprintf('linear: %6.4f\npchip: %6.4f\nspline: %6.4f\nnearest: %6.4f\n', lCp,pCp,sCp,nCp);  

File: python/ch02/ex234.py
from scipy.interpolate import interp1d, PchipInterpolator, CubicSpline

T_cp = np.arange(373, 874, 100)
Cp_data = np.array([29.189, 29.291, 29.462, 29.678, 29.971, 30.269])
target_T = 580

lCp = interp1d(T_cp, Cp_data, kind='linear')(target_T)
pCp = PchipInterpolator(T_cp, Cp_data)(target_T)
sCp = CubicSpline(T_cp, Cp_data)(target_T)
nCp = interp1d(T_cp, Cp_data, kind='nearest')(target_T)

print(f"linear: {lCp:.4f}\npchip: {pCp:.4f}\nspline: {sCp:.4f}\nnearest: {nCp:.4f}")